x-pack/legacy/plugins/apm/scripts/upload-telemetry-data/index.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fs from 'fs';
+import path from 'path';
+import Ocktokit from '@octokit/rest';
+import { merge, chunk, flatten, pick, identity } from 'lodash';
+import axios from 'axios';
+import yaml from 'js-yaml';
+import { Client } from 'elasticsearch';
+import { argv } from 'yargs';
+import { promisify } from 'util';
+import { Logger } from 'kibana/server';
+// @ts-ignore
+import consoleStamp from 'console-stamp';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { CollectTelemetryParams } from '../../../../../plugins/apm/server/lib/apm_telemetry/collect_data_telemetry';
+import { downloadTelemetryMapping } from './download-telemetry-mapping';
+import mapping from '../../mappings.json';
+import { generateSampleDocuments } from './generate-sample-documents';
+
+consoleStamp(console, '[HH:MM:ss.l]');
+
+const githubToken = process.env.GITHUB_TOKEN;
+
+const kibanaConfigDir = path.join(__filename, '../../../../../../../config');
+const kibanaDevConfig = path.join(kibanaConfigDir, 'kibana.dev.yml');
+const kibanaConfig = path.join(kibanaConfigDir, 'kibana.yml');
+
+const xpackTelemetryIndexName = 'xpack-phone-home';
+
+const config = {
+  'apm_oss.transactionIndices': 'apm-*',
+  'apm_oss.metricsIndices': 'apm-*',
+  'apm_oss.errorIndices': 'apm-*',
+  'apm_oss.spanIndices': 'apm-*',
+  'apm_oss.onboardingIndices': 'apm-*',
+  'apm_oss.sourcemapIndices': 'apm-*',
+  'elasticsearch.hosts': 'http://localhost:9200',
+  ...(yaml.safeLoad(
+    fs.readFileSync(
+      fs.existsSync(kibanaDevConfig) ? kibanaDevConfig : kibanaConfig,
+      'utf8'
+    )
+  ) as {}),
+  ...(pick(
+    {
+      'elasticsearch.username': process.env.ELASTICSEARCH_USERNAME,
+      'elasticsearch.password': process.env.ELASTICSEARCH_PASSWORD,
+      'elasticsearch.hosts': process.env.ELASTICSEARCH_HOST
+    },
+    identity
+  ) as {
+    'elasticsearch.username': string;
+    'elasticsearch.password': string;
+  })
```

Comment:
Perhaps it would be cleaner with `safeLoad` done separately?
Also: shouldn't `elasticsearch.hosts` be in the explicit type?

Reply:
I'll split it up a bit. `elasticsearch.hosts` is already in the type because it has a default value, but it doesn't hurt to have it in both, I'll add it there as well (it looks odd).